File: backend/app/schemas/runtime.py
from pydantic import BaseModel, Field


class PostgresConfig(BaseModel):
    host: str = Field(default="127.0.0.1", min_length=1, max_length=255)
    port: int = Field(default=5432, ge=1, le=65535)
    user: str = Field(default="postgres", min_length=1, max_length=255)
    password: str | None = Field(default=None, max_length=512)
    database: str = Field(default="tender", min_length=1, max_length=255)


class RedisConfig(BaseModel):
    host: str = Field(default="127.0.0.1", min_length=1, max_length=255)
    port: int = Field(default=6379, ge=1, le=65535)
    password: str | None = Field(default=None, max_length=512)
    db: int = Field(default=0, ge=0, le=15)


class MinioConfig(BaseModel):
    endpoint: str = Field(default="localhost:9000", min_length=1, max_length=255)
    access_key: str = Field(default="minioadmin", min_length=1, max_length=255)
    secret_key: str | None = Field(default=None, max_length=512)
    bucket: str = Field(default="tender-files", min_length=1, max_length=255)
    secure: bool = False


class RuntimeSettingsInput(BaseModel):
    postgres: PostgresConfig
    redis: RedisConfig
    minio: MinioConfig


class ComponentHealth(BaseModel):
    ok: bool
    message: str | None = None


class RuntimeTestResult(BaseModel):
    postgres: ComponentHealth
    redis: ComponentHealth
    minio: ComponentHealth


class RuntimeSettingsRead(BaseModel):
    postgres: PostgresConfig
    redis: RedisConfig
    minio: MinioConfig
    health: RuntimeTestResult
    config_file: str

File: desktop/src/types/runtime.ts
export interface PostgresConfig {
  host: string
  port: number
  user: string
  password: string | null
  database: string
}

export interface RedisConfig {
  host: string
  port: number
  password: string | null
  db: number
}

export interface MinioConfig {
  endpoint: string
  access_key: string
  secret_key: string | null
  bucket: string
  secure: boolean
}

export interface RuntimeSettingsInput {
  postgres: PostgresConfig
  redis: RedisConfig
  minio: MinioConfig
}

export interface ComponentHealth {
  ok: boolean
  message: string | null
}

export type RuntimeHealthMap = {
  postgres: ComponentHealth
  redis: ComponentHealth
  minio: ComponentHealth
}

export interface RuntimeSettingsRead extends RuntimeSettingsInput {
  health: RuntimeHealthMap
  config_file: string
}
